src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -74,10 +108,42 @@ fun BoolQueryBuilder.mustWhenPresent(query: String, value: LocalDate?): BoolQuer
   return this
 }
 
+fun BoolQueryBuilder.must(query: String, value: Any): BoolQueryBuilder {
+  this.must(QueryBuilders.matchQuery(query, value))
+  return this
+}
+
 fun BoolQueryBuilder.mustWhenTrue(predicate: () -> Boolean, query: String, value: String): BoolQueryBuilder {
   value.takeIf { predicate() }?.let {
     this.must(QueryBuilders
         .matchQuery(query, it))
   }
   return this
-}
\ No newline at end of file
+}
+
+fun BoolQueryBuilder.mustMultiMatchKeyword(value: String, vararg query: String): BoolQueryBuilder {
+  this.must().add(multiMatchQuery(value, *query)
+      .analyzer("whitespace")
+  )
+  return this
+}
+
+
+sealed class Result<out T, out M> {
+  data class NoMatch<out T>(val value: T? = null) : Result<T, Nothing>()
+  data class Match<out M>(val matches: M) : Result<Nothing, M>()
+}
+
+inline infix fun <T, M> Result<T, M>.onMatch(block: (Result.Match<M>) -> Nothing): T? {
+  return when (this) {
+    is Result.NoMatch<T> -> value
+    is Result.Match<M> -> block(this)
+  }
+}
+
+private fun BoolQueryBuilder.withDefaults(matchRequest: MatchRequest): BoolQueryBuilder? {
+  return this
+      .mustWhenTrue({ matchRequest.activeSentence }, "currentDisposal", "1")
+      .must("softDeleted", false)
+}
+
```

Comment:
You're building a nice matching DSL here!  And teaching us all about Kotlin goodness. :+1: 
One small observation - I'm not sold on the word "must".  I think I prefer "include" maybe?  But yeah I'm really being picky now...

Reply:
yeah, that is the word Elastic Search uses; they have `must` and `should`